### PR TITLE
[ConvDiff] Clean up python scripts

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
Clean up `python_scripts` folder in the `ConvectionDiffusionApplication`

Please mark the PR with appropriate tags: 
- Cleanup
- Applications
- Deprecated
- Legacy
- Bugfix

**Changelog**
- Remove the old solvers that are not analysis stage compatible: @RiccardoRossi checking the log I realized that you implemented all of them so please confirm that we don't need them any more.
- Minor fix in the base and transient solver. Now the user can specify only a few variables of the `convection_diffusion_variables` dictionary. Defaults are set for all the non user-defined variables. Deprecated warnings have been removed too.
- Since the `SourceTermTest` used one of the deprecated solvers it required to be adapted to the standard transient solver.
- Rename the `convection_diffusion_base_solver.py` to `convection_diffusion_solver.py` to follow the other Kratos applications standards

### CODE REVIEW

**Review by loumalouomega — APPROVED** [review]
OK

**Review by loumalouomega — CHANGES_REQUESTED** [review]
There are conflicts and the Ci does not pass

**Comment by rubenzorrilla** [comment]
Just pinging all the potential users that might be affected by these changes that come to my mind @KratosMultiphysics/altair @riccardotosi @juliobarna @pablobecker

**Comment by rubenzorrilla** [comment]
> There are conflicts and the Ci does not pass

Conflicts are solved now.

**Review by loumalouomega — DISMISSED** [review]

File: applications/ConvectionDiffusionApplication/tests/source_term_test.py [hunk 1]
```diff
@@ -32,16 +38,45 @@ def __init__(self):
         self.ux = 0.0
         self.source = 0.0
 
+"""Derived convection-diffusion analysis stage to set the test material properties and boundary conditions."""
```
Comment (riccardotosi) [inline applications/ConvectionDiffusionApplication/tests/source_term_test.py]: Doxygen of the class should be below the class definition, see [Doxygen manual](https://www.doxygen.nl/manual/docblocks.html#pythonblocks).

**Comment by riccardotosi** [comment]
Ok to me.

File: applications/ConvectionDiffusionApplication/tests/source_term_test.py (hunk 1 above)
Comment (rubenzorrilla) [reply applications/ConvectionDiffusionApplication/tests/source_term_test.py]: Changing. Thanks for pointing out.

**Review by riccardotosi — DISMISSED** [review]

**Comment by ddiezrod** [comment]
We are not using ConDiff app in altair so far, so ok from my side ;)

**Comment by rubenzorrilla** [comment]
If there are no complaints I'll merge this during next week.

File: applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py [hunk 2]
```diff
@@ -1,133 +0,0 @@
-from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
-# importing the Kratos Library
-from KratosMultiphysics import *
-from KratosMultiphysics.ConvectionDiffusionApplication import *
```
Comment (RiccardoRossi) [inline applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py:4]: is there a BFECC solver which is working?

if not this one which?

**Review by RiccardoRossi — DISMISSED** [review]
i am ok apart for the question about BFECC

File: applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py (hunk 2 above)
Comment (rubenzorrilla) [reply applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py:4]: There is no BFECC solver working... In any case I think it shouldn't be difficult to integrate the BFECC convection in the current solvers.

**Comment by philbucher** [comment]
should this go in before the release?

**Comment by rubenzorrilla** [comment]
> should this go in before the release?

I think we can include it.

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/altair Shall we consider this in our merge before release?

**Comment by rubenzorrilla** [comment]
> @KratosMultiphysics/altair Shall we consider this in our merge before release?

@ddiezrod answered saying that you're not using the `ConvectionDiffusionApplication` so as far as I understand this shouldn't affect you.

File: applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py (hunk 2 above)
Comment (rubenzorrilla) [reply applications/ConvectionDiffusionApplication/python_scripts/bfecc_convection_diffusion_solver.py:4]: Resolving as the semiimplicit solver has been already merged.

**Comment by rubenzorrilla** [comment]
This finally cleans up the `python_scripts` folder of the `ConvectionDiffusionApplication`. There are inconsistencies in the solvers wrapper as the steady solvers require to specify the implicit time integration scheme to avoid throwing a warning. Even though this has no affectation in the results, it might be misleading for the user.

Taking into account that this PR is large enough and that only includes clean up, I'll take care of it in a subsequent PR.

**Review by RiccardoRossi — APPROVED** [review]
